### PR TITLE
[flutter_tools] try deleting the web cache directory before copying new web sdk

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Delete the destination directory for the web sdk before copying the new contents. This might fix an error where the OS refuses to overwrite the files on dev

### CODE REVIEW

File: packages/flutter_tools/lib/src/cache.dart [hunk 1]
```diff
@@ -1556,7 +1564,11 @@ class ArtifactUpdater {
         _ensureExists(tempFile.parent);
         final IOSink ioSink = tempFile.openWrite();
         await _download(url, ioSink);
+        await ioSink.flush();
         await ioSink.close();
+        if (!tempFile.existsSync()) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/cache.dart:1569]: this is to try and surface issues if we download the file incorrectly

File: packages/flutter_tools/lib/src/cache.dart [hunk 2]
```diff
@@ -706,22 +710,26 @@ class FlutterWebSdk extends CachedArtifact {
...
-    } else if (globals.platform.isLinux) {
+    } else if (_platform.isLinux) {
       platformName += 'linux-x64';
-    } else if (globals.platform.isWindows) {
+    } else if (_platform.isWindows) {
       platformName += 'windows-x64';
     }
     final Uri url = Uri.parse('${cache.storageBaseUrl}/flutter_infra/flutter/$version/$platformName.zip');
+    if (location.existsSync()) {
+      location.deleteSync(recursive: true);
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/cache.dart:722]: Should this ignore exceptions?

File: packages/flutter_tools/lib/src/cache.dart [hunk 3]
```diff
@@ -1556,7 +1564,11 @@ class ArtifactUpdater {
         _ensureExists(tempFile.parent);
         final IOSink ioSink = tempFile.openWrite();
         await _download(url, ioSink);
+        await ioSink.flush();
         await ioSink.close();
+        if (!tempFile.existsSync()) {
+          throw Exception();
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/cache.dart:1570]: The exception message will get printed below by the `printTrace()`. You remember what this is for now, but what about in a few  weeks... =)

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/cache.dart:1570]: Good point :)

File: packages/flutter_tools/lib/src/cache.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/cache.dart:722]: I think if we fail to delete this, we'd also likely fail to copy files over it

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
google testing pls

**Comment by jonahwilliams** [comment]
3 hours is long enough!

**Comment by jonahwilliams** [comment]
FYI @renyou it got stuck :(

**Comment by renyou** [comment]
Sorry our service had an upgrade this afternoon and a few things haven't been very stable :(

**Pull request merged**